tests/network/vmi_istio.go

```diff
@@ -0,0 +1,310 @@
+			Pod: &v1.PodNetwork{},
+		},
+	}
+
+	Context("Virtual Machine with masquerade interface and explicitly specified ports", func() {
+		verifyClientServerConnectivity := func(serverVMI *v1.VirtualMachineInstance, targetPort int) (*batchv1.Job, error) {
+			By("Starting HTTP Server")
+			tests.StartHTTPServer(vmi, targetPort)
+
+			By("Getting back the VMI IP")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			vmiIP := vmi.Status.Interfaces[0].IP
+
+			By("Running job to send a request to the server")
+			return virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Create(
+				context.Background(),
+				generateBatchJobSpec(vmiIP, fmt.Sprintf("%d", targetPort), tests.NewHelloWorldJobHTTP),
+				metav1.CreateOptions{},
+			)
+		}
+		BeforeEach(func() {
+			libnet.AddLabelToNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel, "enabled")
+			ports := []v1.Port{
+				{Port: svcDeclaredTestPort},
+				{Port: svcUndeclaredTestPort},
+			}
+			vmi = newRandomCirrosVMIWithIstioSidecarAndDefaultServiceAccountAndNetworkAndInterface(
+				vmiAppSelector,
+				network,
+				ports,
+			)
+			service = newServiceWithAppSelector(
+				fmt.Sprintf("%s-service", vmi.Name),
+				vmi.Namespace,
+				vmiAppSelector,
+			)
+
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			By("Creating VMI and associated Service")
+			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		})
+		AfterEach(func() {
+			libnet.RemoveLabelFromNamespace(tests.NamespaceTestDefault, istioInjectNamespaceLabel)
+			Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.Name, &metav1.DeleteOptions{})).ShouldNot(HaveOccurred())
+			Expect(virtClient.CoreV1().Services(tests.NamespaceTestDefault).Delete(context.TODO(), service.Name, metav1.DeleteOptions{})).ShouldNot(HaveOccurred())
+		})
+		Describe("Inbound traffic", func() {
+			var job *batchv1.Job
+			AfterEach(func() {
+				var backgroundPropagation = metav1.DeletePropagationBackground
+				Expect(virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Delete(context.TODO(), job.Name, metav1.DeleteOptions{PropagationPolicy: &backgroundPropagation})).ShouldNot(HaveOccurred())
+			})
+			table.DescribeTable("request to VMI", func(port int) {
+				job, err = verifyClientServerConnectivity(vmi, port)
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
```

Comment:
How do we know for sure that envoy was involved? The fact that with peer authentication it doesn't work?

Reply:
Yes